src/utils/supergraph_config.rs

```diff
@@ -116,28 +119,35 @@ fn merge_supergraph_configs(
     local_config: Option<SupergraphConfig>,
     target_federation_version: Option<&FederationVersion>,
 ) -> Option<SupergraphConfig> {
+    // We use the federation version explicitly given at the command line as top
+    // priority; otherwise the version explicitly given in the local config
+    // file; otherwise the version fetched from Studio.
+    let resolved_federation_version = target_federation_version
+        .cloned()
+        .or(local_config
+            .as_ref()
+            .and_then(|it| it.get_federation_version()))
+        .or(remote_config
+            .as_ref()
+            .and_then(|it| it.get_federation_version()))
+        .unwrap_or(FederationVersion::LatestFedTwo);
```

Comment:
probably want `.or_else` so that what's inside each `.or()` isn't evaluated (`.or_else` is lazily evaluated)